src/arithmetic/vector_funcs/vector_funcs.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright FalkorDB Ltd. 2023 - present
+ * Licensed under the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "RG.h"
+#include "../../value.h"
+#include "../func_desc.h"
+#include "../../errors.h"
+#include "../../util/arr.h"
+#include "../../datatypes/array.h"
+#include "../../datatypes/vector.h"
+
+// create a vector from an array of floats
+// vector32f([0.2, 0.12, 0.3178])
+SIValue AR_VECTOR32F
+(
+	SIValue *argv,      // arguments
+	int argc,           // number of arguments
+	void *private_data  // private data
+) {
+	// expecting input to be an array of 32 bit floats
+	// vector32f([0.2, 0.12, 0.3178])
+
+	SIValue arr = argv[0];
+
+	// return NULL if input is NULL
+	SIType t = SI_TYPE(arr);
+	if(t == T_NULL) {
+		return SI_NullVal();
+	}
+
+	ASSERT(t == T_ARRAY);
+
+	// create a vector of the same length as the input array
+	uint32_t n         = SIArray_Length(arr);
+	SIValue  v         = SI_Vector32f(n);
+	float    *elements = (float*)SIVector_Elements(v);
+
+	// validate input contains only floats
+	// save each float into the vector's internal values array
+	for(uint32_t i = 0; i < n; i++) {
+		SIValue v = SIArray_Get(arr, i);
+
+		// encountered a non-numeric value
+		if(!(SI_TYPE(v) & SI_NUMERIC)) {
+			SIValue_Free(v);
```

Comment:
can we use SIArray_AllOfType in the beginning?

Reply:
Yes, I think this is better, although it will double the number of iterations.